server/market/orderrouter.go

```diff
@@ -377,10 +441,97 @@ func (r *OrderRouter) handleMarket(user account.AccountID, msg *msgjson.Message)
 		req:   market,
 		msgID: msg.ID,
 	}
-	if err := tunnel.SubmitOrder(oRecord); err != nil {
-		log.Warnf("Market failed to SubmitOrder: %v", err)
-		return msgjson.NewError(msgjson.UnknownMarketError, "failed to submit order")
-	}
+
+	checkCoins := func() (tryAgain bool, msgErr *msgjson.Error) {
```

Comment:
There is a fair amount of code duplication between `handleLimit` and `handleMarket`, but in trying to refactor it out, I ended up with functions with an excessive number of inputs, plus ugly error handling.  These closures actually simplify things greatly.  The main differences are the `reqVal` (required order funds) checking.